src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/MailingSleepIntervalProvider.java

```diff
@@ -0,0 +1,58 @@
+package pl.tomaszdziurko.jvm_bloggers.mailing;
+
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.PRODUCTION;
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.STAGE;
+
+@Component
+@Slf4j
+public class MailingSleepIntervalProvider {
+
```

Comment:
So the methods should also be renamed to have nothing to do with mailing IMHO. 

>if we start using it in another functionality, this name will become misleading

That's why it will be better to just remove mailing keyword from the class and method names.